lib/whitehall_importer/migrate_assets.rb

```diff
@@ -33,15 +33,26 @@ def migrate_asset(whitehall_asset)
 
     def redirect_asset(whitehall_asset)
       GdsApi.asset_manager.update_asset(
-        whitehall_asset.whitehall_asset_id,
+        asset_id(whitehall_asset),
         redirect_url: whitehall_asset.content_publisher_asset.file_url,
       )
       whitehall_asset.update!(state: "redirected")
     end
 
     def remove_asset(whitehall_asset)
-      GdsApi.asset_manager.delete_asset(whitehall_asset.whitehall_asset_id)
+      GdsApi.asset_manager.delete_asset(asset_id(whitehall_asset))
       whitehall_asset.update!(state: "removed")
     end
+
+    def asset_manager_attributes(whitehall_asset)
+      @asset_manager_attributes ||= GdsApi.asset_manager
+                                          .whitehall_asset(whitehall_asset.legacy_url_path)
+                                          .to_hash
+    end
+
+    def asset_id(whitehall_asset)
+      url = asset_manager_attributes(whitehall_asset)["id"]
+      url[/\/assets\/(.*)/, 1]
```

Comment:
Would you prefer this method to me the same as as in `models/file_attachment_asset.rb`?

Reply:
I've changed to split the ID into an array, and then grab the last element, which works.  Does that look ok to you?

Reply:
Yup looks good, I don't really mind which way we do it - just so that we don't approach the same problem differently in two different places. 

It's in that tricky realm of such a small problem that a helper function seems over the top but then two different approaches makes it look like they're solving different scenarios.

I guess the actual solution to all of this is to put the actual id into the asset manager API output at some point and end this madness, but alas we'd have to give it a silly name like "actual_id" to not break everything else so feels a job for the future. I'll add it to my things to fix on GOV.UK board.